builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -599,7 +600,11 @@ func (p *Provisioner) runChefClientFunc(
 
 		// Policyfiles do not support chef environments, so don't pass the `-E` flag.
 		if p.UsePolicyfile {
-			cmd = fmt.Sprintf("%s -j %q", chefCmd, fb)
+			if p.NamedRunList == "" {
+				cmd = fmt.Sprintf("%s -j %q", chefCmd, fb)
+			} else {
+				cmd = fmt.Sprintf("%s -j %q -n %q", chefCmd, fb, p.NamedRunList)
+			}
 		} else {
 			cmd = fmt.Sprintf("%s -j %q -E %q", chefCmd, fb, p.Environment)
 		}
```

Comment:
I think it would be more idiomatic Go (and a little cleaner to read) to write this as follows:

```go
// Policyfiles do not support chef environments, so don't pass the `-E` flag.
switch {
case p.UsePolicyfile && p.NamedRunList == "":
  cmd = fmt.Sprintf("%s -j %q", chefCmd, fb)
case p.UsePolicyfile && p.NamedRunList != "":
  cmd = fmt.Sprintf("%s -j %q -n %q", chefCmd, fb, p.NamedRunList)
default:			}
  cmd = fmt.Sprintf("%s -j %q -E %q", chefCmd, fb, p.Environment)
}
```